### DISCUSSION TITLE
Issue with RBAC data access

### DISCUSSION THREAD

**Discussion by SantoshSrinivas79** [Budibase/budibase#3367] [Help]
I made a simple app to test out automation and role-based access. The roles don't seem to be getting applied in the data access.

tldr: a user with `basic` role is able to see data that required `admin` access

A quick recording of the observation is shown below and also attached is the export file.

Please clarify if I've understood RBAC incorrectly. Thx!

https://www.loom.com/share/9b505b36ceac48f887ffa7c08e15ca2c

[Permissions-export-1636854071524.txt](https://github.com/Budibase/budibase/files/7533031/Permissions-export-1636854071524.txt)

**Comment by itsamemarkus** [comment]
I had the same experience.. This is just a wild guess: Maybe it's connected to https://github.com/Budibase/budibase/issues/3215 so that it tries to lookup RBAC from the users table within an app, which is usually empty except for the builder account.

Would be interested in a dev clarification as well.

**Comment by mike12345567** [comment]
Hi @SantoshSrinivas79 - thanks for reporting this, that does appear to be an issue, it appears to be ignoring the assigned permission and just falling back to the base permissions (all tables start out by default with a `BASIC` permission set) - i'll get a fix pushed out for this asap.

**Comment by SantoshSrinivas79** [comment]
Hi @mike12345567 ... I believe you have pushed a fix to the cloud platform. Correct? I tried and it seems to work fine now.

Any idea when the self-host docker image will be updated?

Also, while on the topic of RBAC, can you pls take a look at [this](https://github.com/Budibase/budibase/discussions/3368#discussioncomment-1644908) issue?

I am trying to figure out if this can be accomplished but can't quite it out. I need a user-specific private key `User Secret` that cannot be accessed by anyone else in the same or lower role. Even with the API. Is this possible?

Thank you!

**Reply by shogunpurple** [reply]
Hi @SantoshSrinivas79 - we will update the self host images this morning. 

I will have a look at your other issue now too.

**Reply by SantoshSrinivas79** [reply]
Hi @shogunpurple, thank you for bearing with me on this. The bugs still seem to be there. The RBAC seems to be working for public users. However, a `BASIC` user is able to `READ` data which is enabled only for `ADMIN`.

My app on the cloud is: https://mypad.budibase.app/app_mypad_e78726b168164294b15f122d3b5c7f59#/applicants

Pls feel free to play around with it if you have access to it
